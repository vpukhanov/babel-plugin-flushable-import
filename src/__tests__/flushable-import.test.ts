import 'regenerator-runtime/runtime'

import flushableImport from '../flushable-import'

const config = {
  load: () => import('../flushable-import'),
  resolve: () => null,
  path: () => '<path>',
  chunkName: '<chunkName>',
}

test('returns given config', () => {
  expect(flushableImport(config)).toMatchSnapshot()
})

test.each(['then', 'catch'])('adds %s field to config', fieldName => {
  expect(flushableImport(config)).toHaveProperty(fieldName)
})

test('awaits to imported module', async () => {
  const expectedMod = await import('../flushable-import')
  const mod = await flushableImport(config)
  expect(mod).toEqual(expectedMod)
})
