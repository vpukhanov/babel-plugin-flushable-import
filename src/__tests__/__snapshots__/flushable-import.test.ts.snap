// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns given config 1`] = `
Object {
  "catch": [Function],
  "chunkName": "<chunkName>",
  "load": [Function],
  "path": [Function],
  "resolve": [Function],
  "then": [Function],
}
`;
